sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsBufferStore.scala

```diff
@@ -304,11 +305,48 @@ abstract class RapidsBufferStore(
       }
     }
 
+    override def getMemoryBuffer: MemoryBuffer = materializeMemoryBuffer
+
+    override def getDeviceMemoryBuffer: DeviceMemoryBuffer = {
+      deviceRapidsBuffer match {
+        case Some(rapidsBuffer) => rapidsBuffer.getDeviceMemoryBuffer
+        case None =>
+          catalog.acquireBuffer(id, DEVICE) match {
+            case Some(acquiredBuffer) =>
+              deviceRapidsBuffer = Some(acquiredBuffer)
+              deviceRapidsBuffer.get.getDeviceMemoryBuffer
```

Comment:
This needs to close `acquiredBuffer` after calling `getDeviceMemoryBuffer` on it, otherwise we're unnecessarily pinning the device memory even after the caller releases the resulting `DeviceMemoryBuffer`.  It will never be closed until this buffer is closed, meaning unspill becomes a one-way ticket, never to be spilled again.  That's not what we want and will lead to OOM situations that are avoidable.

The `DeviceMemoryBuffer` being wielded by the caller and eventually closed by the caller will prevent the memory from being released as long as it hasn't been closed.  That's why we don't need to track the RapidsDeviceBuffer instance in this buffer instance.